docs/source/supported-types.rst

```diff
@@ -687,6 +688,54 @@ Type checking also still applies.
       File "<stdin>", line 1, in <module>
     msgspec.ValidationError: Expected `str`, got `int` - at `$[1][1]`
 
+``UNSET``
+---------
+
+`msgspec.UNSET` is a singleton object used to indicate that a field has no set
+value. This is useful for cases where you need to differentiate between a
+message where a field is missing and a message where the field is explicitly
+``None``.
+
+.. code-block:: python
+
+    >>> from msgspec import Struct, UnsetType, UNSET, json
+
+    >>> class Example(Struct):
+    ...     x: int
+    ...     y: int | None | UnsetType = UNSET  # a field, defaulting to UNSET
+
+During encoding, any field containing ``UNSET`` is omitted from the message.
+
+.. code-block:: python
+
+    >>> json.encode(Example(1))  # y is UNSET
+    b'{"x":1}'
```

Comment:
Would this additional example be helpful?

```suggestion
    >>> json.encode(Example(1))  # y is UNSET
    b'{"x":1}'
    
    >>> json.encode(Example(1, UNSET))  # y is UNSET
    b'{"x":1}'
```

Reply:
Thanks, added!